### DISCUSSION TITLE
What is the wait_until_pressed equivalent?

### DISCUSSION THREAD

**Discussion by TheWendyPower** [pybricks/support#940] [Q&A] [answered]
For example, the robot performs a series of tasks and comes back home. Then the next section of code will not execute unit a specified button is pressed. With Spike Prime Python it would look like something like:

```
motor_pair.move(15, 'cm')
hub.left_button_wait_until_pressed()
motor_pair.move(75, 'cm')
```

My kids have several times they need to touch or reposition the robot before it needs to perform the next series of tasks. Where only one program can be downloaded to the hub at a time, this would help make sure the student was ready before the robot started moving again.

**Comment by dlech** [comment, marked as answer]
You can wait for the left button to be pressed like this:

```python
while Button.LEFT not in hub.buttons.pressed():
    wait(10)
```

**Reply by TheWendyPower** [reply]
Thanks, that worked as needed.

**Comment by laurensvalk** [comment]
> only one program can be downloaded to the hub at a time,

See also this example: https://github.com/pybricks/support/issues/861

It lets you prepare (and test) several independent programs, which you could then run from your own "main menu" with letters or numbers to indicate the included programs.
